scintillometry/tests/test_psrfits_hdu_read.py

```diff
@@ -0,0 +1,126 @@
+# Licensed under the GPLv3 - see LICENSE
+"""Full-package tests of psrfits reading routine sources."""
+
+import pytest
+import numpy as np
+import os
+import astropy.units as u
+from astropy.time import Time
+
+from ..psrio import open_read
+
+test_data = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+
+
+class TestRead:
+    def setup(self):
+        self.chime_data = os.path.join(test_data, "CHIME_B1937+21_grid01_beam_7"
+                                            "_58464_79458.fits")
+        self.crab_data = os.path.join(test_data, "pulse_169370884.rf")
+
+
+class TestCHIMERead(TestRead):
+    def setup(self):
+        super().setup()
+        self.readers = open_read(self.chime_data)
+
+    def test_shape(self):
+        # The header HUD does not have any data
+        shape0 = self.readers[0].shape
+        assert shape0 == (0,)
```

Comment:
As said above, I'm not sure why `shape` is defined at all, so just check `with pytest.raises(AttributeError)`

Reply:
Since the reader is based on the StreamGenerator, which requires the shape.

Reply:
Perhaps the main header should then not be based on `StreamGenerator`?

Reply:
We can just make a small independent primary reader class for it. But this kind of ties to the question, how to make an integrated reader.